src/queue/bookmark/utils.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  generateWorkflowId,
+  WorkflowQueue,
+  WorkflowTopic,
+  WorkflowTopicScope,
+} from '../common';
+import { BookmarkReminderParams, bookmarkReminderWorkflow } from './workflows';
+import { getTemporalClient } from '../client';
+
+export const getReminderWorkflowId = ({
+  userId,
+  postId,
+  remindAt,
+}: BookmarkReminderParams) =>
+  generateWorkflowId(WorkflowTopic.Bookmark, WorkflowTopicScope.Reminder, [
+    userId,
+    postId,
+    remindAt.toString(),
+  ]);
+
+export const runReminderWorkflow = async (params: BookmarkReminderParams) => {
+  const workflowId = getReminderWorkflowId(params);
```

Comment:
One thing to keep in mind is that the IDs are unique, so we can't trigger the workflow with the same params multiple times. Is that a concern you think?

Reply:
We should be safe here, I included the date (turned to int) as params which will provide the uniqueness of the workflow.